File: HbSolution/src/main/java/com/api/HbSolution/DTO/ProdutoDTO.java
package com.api.HbSolution.DTO;

import java.math.BigDecimal;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class ProdutoDTO {

    private Long id;
    private String nome;
    private String descricao;
    private BigDecimal preco;
    private Integer quantidadeEstoque;
    private String codigoBarras;
    private String categoria;
    private Long idEmpresa; // ID da empresa associada ao produto

}

File: HBSolution-web/src/app/models/dtos/ProdutoDTO.ts
import { BaseDTO } from "./baseDTO";

export class ProdutoDTO extends BaseDTO {
    public nome!: string;
    public descricao!: string;
    public preco!: number;
    public quantidadeEstoque!: number;
    public codigoBarras!: string;
    public categoria!: string;
    public idEmpresa!: number;

    constructor(model?: ProdutoDTO) {
        super(model);
        if (model) {
            this.nome = model.nome;
            this.descricao = model.descricao;
            this.preco = model.preco;
            this.quantidadeEstoque = model.quantidadeEstoque;
            this.codigoBarras = model.codigoBarras;
            this.categoria = model.categoria;
            this.idEmpresa = model.idEmpresa;
        }
    }

    static setPayload(json: any): ProdutoDTO {
        const produto = new ProdutoDTO();
        produto.id = json.id;
        produto.nome = json.nome;
        produto.descricao = json.descricao;
        produto.preco = json.preco;
        produto.quantidadeEstoque = json.quantidadeEstoque;
        produto.codigoBarras = json.codigoBarras;
        produto.categoria = json.categoria;
        produto.idEmpresa = json.idEmpresa;
        produto.createdAt = json.createdAt;
        return produto;
    }
}
